torchtext/datasets/ag_news.py

```diff
@@ -30,11 +31,18 @@
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=4)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def AG_NEWS(root, split):
-    path = download_from_url(URL[split], root=root,
-                             path=os.path.join(root, split + ".csv"),
-                             hash_value=MD5[split],
-                             hash_type='md5')
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def AG_NEWS(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL[split]])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, split + ".csv"),
+        hash_dict={os.path.join(root, split + ".csv"): MD5[split]},
+        hash_type="md5"
+    )
+    cache_dp = HttpReader(cache_dp)
+    cache_dp = cache_dp.end_caching(mode="w", same_filepath_fn=True)
```

Comment:
NIT: can we combine these two lines like we did for [AmazonReviewFull](https://github.com/pytorch/text/pull/1499/files#diff-61c4706569928ac5352feafe9003545f246296ecc93c10e20120747f3707ccbdR52) and [AmazonReviewPolarity](https://github.com/pytorch/text/pull/1490/files#diff-daeb7a9b451f1cb20240d2fda55074b6f03d48849646c931c513050af8eff1a5R46)

We want to have as much consistency between the dataset implementations as possible